### PR TITLE
[WFLY-9753] Pull Request Template

### PR DESCRIPTION
**Pull request by rsvoboda**
JIRA issue - https://issues.jboss.org/browse/WFLY-9753

PR template to promote and ensure best practice for Pull Requests, inspired by Brian's presentation about best practice.

@kabir, @bstansberry, @jamezp would you like to add or remove something from the template?

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+- [ ] Pull Request contains link to the JIRA issue
+- [ ] Pull Request contains description of the issue
+- [ ] Pull Request does not include fixes for issues other than the main ticket
+- [ ] Attached commits represent units of work and are properly formatted
+
+For new features, bigger changes and major/minor component upgrades make sure your PR meets also the following requirements:
+
+ - [ ] Pull Request requires a change to the documentation
+ - [ ] Documentation have been updated accordingly
+ - [ ] Tests were added to cover changes
```
Comment (kabir) [inline .github/PULL_REQUEST_TEMPLATE.md]: It is missing:
Analysis was done
Test Plan has been done
Tests were verified in advance

Comment (kabir) [inline .github/PULL_REQUEST_TEMPLATE.md]: It is missing:
Analysis was done
Test Plan has been done
Tests were verified in advance

**Comment by n1hility** [comment]
Looks good, but I think we should probably mention that if you are not an active contributor of the WildFly project that you can request sponsorship by one of the members to help guide you through the process.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Thanks for submitting your Pull Request!
+
+Please make sure your PR meets the following requirements:
+
+- [ ] Pull Request title is properly formatted: `[WFLY-XYA] Subject`
+- [ ] Pull Request contains link to the JIRA issue
+- [ ] Pull Request contains description of the issue
+- [ ] Pull Request does not include fixes for issues other than the main ticket
```
Comment (darranl) [inline .github/PULL_REQUEST_TEMPLATE.md]: I think moving to PRs representing complete features I think it is going to be unrealistic to expect PRs never contain related fixes.

There are approaches that can be used in git to submit those fixes independently as well but they may be needed in the topic branch being submitted for that branch to build and pass the testsuite.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Thanks for submitting your Pull Request!
+
+Please make sure your PR meets the following requirements:
+
+- [ ] Pull Request title is properly formatted: `[WFLY-XYA] Subject`
```
Comment (rhusar) [inline .github/PULL_REQUEST_TEMPLATE.md]: [ and ] around Jira ID are completely redundant and should not be required neither promoted to be used.

**Review by rhusar — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (rsvoboda) [reply .github/PULL_REQUEST_TEMPLATE.md]: thanks for the comment, some projects use this notation, also in many WildFly commits

We can have this:
[ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`

I personally like [WFLY-XYZ] more :)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (rsvoboda) [reply .github/PULL_REQUEST_TEMPLATE.md]: You can always keep this item not selected and explain more in PR description. You can think about this as questionnaire with input for RC / gate-keeper for easier checks

I believe we should not mix too many thing together, it make back-porting and git bisect complicated

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (rhusar) [reply .github/PULL_REQUEST_TEMPLATE.md]: Sure, but we could also use this opportunity to standardize on the effective form.

Comment (rsvoboda) [reply .github/PULL_REQUEST_TEMPLATE.md]: @kabir ^^^ [ ] or not ?

Comment (kabir) [reply .github/PULL_REQUEST_TEMPLATE.md]: Personally, I don't care one way or another about the brackets. It is very nice of course to have the issue keys in the title. But what I **REALLY** :) care about is having the links in the description so I can click on those

**Pull request merged**